### PR TITLE
Vo ia allow user join existing list

### PR DESCRIPTION
**Pull request by vicowiti**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
Added a form that lets a user join an existing Shopping list and shows an error if the shopping list does not exist.

## Acceptance Criteria
  

- [x] The Home view shows a form that allows the user to enter a token to join an existing list, in addition to the button that allows them to create a new list.

 - [x] The input that accepts the list token has a semantic label element associated with its
 - [x] The user can submit this form with both the mouse and the Enter key
 - [x] If the list exists,
the token is saved in localStorage
the user is redirected to the List view and shown the items on that list
 - [x] If the list does not exist, the user is shown an error message that explains the problem
 - [x] If a user does already have a token:
 -[x] They are automatically redirected to the List view.

 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria

From your terminal, do `git checkout main`  pull down this branch with `git pull origin vo-ia-allow-user-join-existing-list` and check that branch out with
`git checkout vo-ia-allow-user-join-existing-list`
Test the feature by typing in an existing 3 word token in the Join an existing shopping list input tab.  If it exists, it will take you to the list page.
If it doesn’t exist, it will throw an error.

### CODE REVIEW

File: src/views/Home.jsx [hunk 1]
```diff
@@ -1,13 +1,34 @@
+import { useEffect } from 'react';
+import { useState } from 'react';
```
Comment (klezi10) [inline src/views/Home.jsx]: Probably best to merge these in one line

**Review by klezi10 — APPROVED** [review]
Well done! Just a note to remove all the console logs and the commented out console logs.

**Review by michhub-dev — CHANGES_REQUESTED** [review]
Nice work. Remember to remove the console.log and commented codes before merging.

File: src/App.jsx [hunk 2]
```diff
@@ -23,6 +24,8 @@ export function App() {
 	 * have tokens), and use `setListToken` when you allow a user
 	 * to create and join a new list.
 	 */
+	const [formToken, setFormToken] = useState('');
+	// const [errorMessage, setErrorMessage] = useState('');
```
Comment (michhub-dev) [inline src/App.jsx]: Like this one, if not in use.

**Review by michhub-dev — APPROVED** [review]

**Review by michhub-dev — APPROVED** [review]

**Review by Amaka202 — COMMENTED** [review]
Good job team🥳🥳🥳 @mide358  @vicowiti 

Everything works as expected and the code is very clear and readable. To improve user experience, maybe you can account for when the user didn't input any token and tries to send the form. Now it just doesn't do anything, I suggest you either disable the button until they type something in the input, or show a message telling them to type a token before submitting.

Asides from that, everything works as expected. Well done✅✅✅

**Comment by mide358** [comment]
> Good job team🥳🥳🥳 @mide358 @vicowiti
> 
> Everything works as expected and the code is very clear and readable. To improve user experience, maybe you can account for when the user didn't input any token and tries to send the form. Now it just doesn't do anything, I suggest you either disable the button until they type something in the input, or show a message telling them to type a token before submitting.
> 
> Asides from that, everything works as expected. Well done✅✅✅

Hello @Amaka202 . Thanks for the heads up, the button has been disabled .

**Review by Amaka202 — APPROVED** [review]

**Pull request merged**